packages/bindgen/src/templates/browser.ts

```diff
@@ -0,0 +1,897 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+import { strict as assert } from "assert";
+
+import { TemplateContext } from "../context";
+import { CppVar, CppFunc, CppFuncProps, CppCtor, CppMethod, CppClass, CppDecls } from "../cpp";
+import {
+  bindModel,
+  BoundSpec,
+  Class,
+  InstanceMethod,
+  StaticMethod,
+  Property,
+  Type,
+  Primitive,
+  Pointer,
+  Template,
+} from "../bound-model";
+
+import { doJsPasses } from "../js-passes";
+
+const emscripten_call_args = new CppVar("const emscripten::val", "args");
+
+function tryWrap(body: string) {
+  return `try {
+                ${body}
+            } catch (const std::exception& ex) {
+                emscripten::val(ex.what()).throw_();
```

Comment:
We shouldn't be throwing raw strings as errors, but that can be changed to use Core's unified exception handling later.

Reply:
I'll add a TODO to use unified Core exception 👍